cpp/src/arrow/pretty_print.h

```diff
@@ -77,6 +77,9 @@ struct PrettyPrintOptions {
 
   /// If true, display schema metadata when pretty-printing a Schema
   bool show_schema_metadata = true;
+
+  /// Delimiter for separating individual elements of an Array (e.g. ",")
+  std::string array_element_delimiter = ",";
```

Comment:
Correct. From my understanding this should be relatively straightforward if we still want to change the default display.

Reply:
Marking this as resolved since we can address https://github.com/apache/arrow/issues/30951 in a follow up PR by changing the default value of `array_element_delimiter` (or whatever name we decide on using).